src/lib.rs

```diff
@@ -313,12 +312,34 @@ where
     }
 
     fn poll_close(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        match (*self).with_context(Some((ContextWaker::Write, cx)), |s| s.close(None)) {
-            Ok(()) => Poll::Ready(Ok(())),
-            Err(::tungstenite::Error::ConnectionClosed) => Poll::Ready(Ok(())),
-            Err(err) => {
-                debug!("websocket close error: {}", err);
-                Poll::Ready(Err(err))
+        if self.closing {
```

Comment:
Should other operations on the `WebSocketStream` maybe also check for this in the beginning and return an error directly?

Reply:
I'm not sure, I believe the underlying `tungstenite` implementation will return an error as appropriate, so maybe we can save the conditional branching and keep code simpler here?